src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsControl.cs

```diff
@@ -1046,18 +1046,20 @@ protected virtual void PrepareContainerForItemOverride(DependencyObject element,
 
 			var styleFromItemsControl = ItemContainerStyle ?? ItemContainerStyleSelector?.SelectStyle(item, element);
 
-			object GetContent()
+			void SetContent(UIElement container, DependencyProperty contentProperty)
 			{
 				var displayMemberPath = DisplayMemberPath;
 				if (string.IsNullOrEmpty(displayMemberPath))
 				{
-					return item;
+					container.SetValue(contentProperty, item);
 				}
 				else
 				{
-					// TODO: Cache the BindingPath
-					var b = new BindingPath(displayMemberPath, item) { DataContext = item };
-					return b.Value;
+					container.SetBinding(contentProperty, new Binding
```

Comment:
@jeromelaban Should this be checking if the binding already exists somehow?

Reply:
In general, it's best to validate if there's one with `GetBindingExpression`, particularly if we're reusing the container.

Reply:
That being said, we're clearing those `DataContext` and `Content` in `CleanUpContainer`, though we're clearing the bindings properly on `Clear()` so we may have duplicates there.

Reply:
Adjusted DependencyPropertyDetails to only keep a single binding, and dispose previous bindings.